data-tx/src/main/java/io/micronaut/transaction/support/DefaultTransactionDefinition.java

```diff
@@ -170,10 +170,12 @@ public final Isolation getIsolationLevel() {
      */
     public final void setTimeout(@NonNull Duration timeout) {
         //noinspection ConstantConditions
-        if (timeout == null || timeout.isNegative()) {
+        if (timeout == null || (timeout.isNegative() && TransactionDefinition.TIMEOUT_DEFAULT.getSeconds() != timeout.getSeconds())) {
```

Comment:
This was needed since changes in annotation to pull default values, which is -1 and before that change this was null so need to add additional check to avoid exception